test/basic/optimalisationCheck.wast

```diff
@@ -0,0 +1,50 @@
+;; tests for Walrus where optimalisation shouldn't occure
+
+(module
+
+  ;; in the following tests there will be an EqualZero, then
+  ;; a JumpIfTrue/JumpIfFalse that should not be unified
+
+  (func (export "test1") (param $input i32) (result i32)
+    local.get $input
+    i32.eqz
+    i32.const 0
+    br_if 0
+  )
+
+  (func (export "test2") (param $input i32) (result i32)
+    local.get $input
+    i32.eqz
+    i32.const 0
+    if
+      nop
+    else
+      nop
+    end
```

Comment:
There should be a variant of this code, where there is a branch after `end`

Reply:
You mean a `br 0`, right? Could you explaine me why, please?

Reply:
This test does not test functional behaviour depending on eqz, so there should be some change depending on it. But my description was poor. The point is, nop/nop does nothing, there should be something which has an effect, which can be compared to something.

Reply:
Okay.